### PR TITLE
Implemented the tag values iterator for `SHOW TAG VALUES`

### LINKED ISSUE #5593: [0.10.0] SHOW TAG VALUES does not work on cluster
`SHOW TAG VALUES` does not return the correct results from all nodes in a fully replicated cluster (default config values).

Node 1:

```
> create database db
> use db
Using database db
> insert cpu,tk=tv value=1
> show tag values with key = "tk"
name: tkTagValues
-----------------
tk
tv

>
```

Node 2 (after all commands on node 1 have completed):

```
> show tag values with key = "tk"
name: tkTagValues
-----------------
tk
tv

>
```

Node 3 (after all commands on node 1 have completed):

```
> show tag values with key = "tk"
>
```

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #5593.

### CODE REVIEW

**Comment by jsternberg** [comment]
I had to add another filtering function which was a bit weird and I would prefer another solution with more code reuse, but I couldn't think of a way that didn't break another part of functionality.

**Comment by jsternberg** [comment]
Not quite done yet. It still needs to handle more functionality that I didn't realize needed to be covered. This kind of call has to be supported: `SHOW TAG VALUES WITH KEY = host WHERE region =~ /ca.*/`

**Comment by jsternberg** [comment]
I think it's working now.

@jwilder @e-dard @benbjohnson please review. I'm not sure how happy I am with the end result, but I do believe it works. This change also affects backwards compatibility so I would like any feedback regarding that. The change to the output is located in the commit message.

File: influxql/ast.go [hunk 1]
```diff
@@ -3397,7 +3397,11 @@ func Rewrite(r Rewriter, node Node) Node {
 		n.Fields = Rewrite(r, n.Fields).(Fields)
 		n.Dimensions = Rewrite(r, n.Dimensions).(Dimensions)
 		n.Sources = Rewrite(r, n.Sources).(Sources)
-		n.Condition = Rewrite(r, n.Condition).(Expr)
+		if cond := Rewrite(r, n.Condition); cond != nil {
+			n.Condition = cond.(Expr)
+		} else {
+			n.Condition = nil
+		}
```
Comment (benbjohnson) [inline influxql/ast.go]: Can you make a `RewriteExpr()` that accepts and returns an `Expr` so that you don't have to do the `nil` cast everywhere?

Comment (jsternberg) [inline influxql/ast.go]: That sounds like a good idea. I can do that.

File: influxql/statement_rewriter.go [hunk 2]
```diff
@@ -149,3 +151,88 @@ func rewriteShowTagKeysStatement(stmt *ShowTagKeysStatement) (Statement, error)
...
+			}
+
+			// Set condition or "AND" together.
+			if condition == nil {
+				condition = expr
+			} else {
+				condition = &BinaryExpr{Op: AND, LHS: expr, RHS: condition}
+			}
+		}
+	}
```
Comment (benbjohnson) [inline influxql/statement_rewriter.go]: I refactored this section out in my `SHOW SERIES` branch. You can pull it out if you want to reuse it:

https://github.com/benbjohnson/influxdb/blob/5997efa5fbbf000d17e8a5e81782d9ff2b2d72ec/influxql/statement_rewriter.go#L123-L172

File: influxql/statement_rewriter.go [hunk 3]
```diff
@@ -149,3 +151,88 @@ func rewriteShowTagKeysStatement(stmt *ShowTagKeysStatement) (Statement, error)
...
+						LHS: &VarRef{Val: "_tagKey"},
+						RHS: &StringLiteral{Val: tagKey},
+					},
+				}
+			} else {
+				expr = &BinaryExpr{
+					Op:  EQ,
+					LHS: &VarRef{Val: "_tagKey"},
+					RHS: &StringLiteral{Val: tagKey},
+				}
```
Comment (benbjohnson) [inline influxql/statement_rewriter.go]: Can you create the `_tagKey = value` expression above the `if` so you can reuse it in the `if` & `else` blocks?

File: tsdb/meta.go [hunk 4]
```diff
@@ -142,16 +142,16 @@ func (d *DatabaseIndex) TagsForSeries(key string) map[string]string {
 	return ss.Tags
 }
 
-// measurementsByExpr takes and expression containing only tags and returns
+// measurementsByExpr takes an expression containing only tags and returns
 // a list of matching *Measurement.
-func (d *DatabaseIndex) measurementsByExpr(expr influxql.Expr) (Measurements, error) {
+func (d *DatabaseIndex) measurementsByExpr(expr influxql.Expr) (Measurements, bool, error) {
```
Comment (benbjohnson) [inline tsdb/meta.go]: Can you document the `bool` return variable?

**Comment by benbjohnson** [comment]
Overall it lgtm. Can you post a before/after example of the output and verify with @pauldix about the change. I'm ok with it.

**Comment by jsternberg** [comment]
Change in output from:

```
> SHOW TAG VALUES WITH KEY IN (host, region)
name: hostTagValues
-------------------
host
server01
server02

name: regionTagValues
---------------------
region
useast
```

to

```
> SHOW TAG VALUES WITH KEY IN (host, region)
name: cpu
---------
key     value
host    server01
region  useast

name: mem
---------
key     value
host    server02
region  useast
```

**Comment by pauldix** [comment]
new output format looks good to me. Does it support `SHOW TAG VALUES FROM cpu WITH KEY IN (host, region)`?

@mark-rushakoff: this change will almost certainly break Chronograf so we'll have to update for it.

**Comment by jsternberg** [comment]
@pauldix yep, it supports that.

```
> insert cpu,host=server01 value=2
> insert mem,host=server02 value=3
> show tag values from cpu with key = host
name: cpu
---------
key     value
host    server01

> show tag values with key = host
name: cpu
---------
key     value
host    server01


name: mem
---------
key     value
host    server02
```

**Pull request merged**